compiler/semantic/scope.go

```diff
@@ -45,6 +45,18 @@ func (s *Scope) DefineVar(name string) error {
 	return nil
 }
 
+func (s *Scope) DefineAs(name string) error {
+	b := s.tos()
+	if _, ok := b.symbols[name]; ok {
+		return fmt.Errorf("symbol %q redefined", name)
+	}
+	ref := &dag.This{Kind: "This"}
+	// We add the symbol to the table but don't bump nvars because
+	// it's not a var and doesn't take a slot in the batch vars.
+	b.Define(name, ref)
```

Comment:
```suggestion
	// We add the symbol to the table but don't bump nvars because
	// it's not a var and doesn't take a slot in the batch vars.
	b.Define(name, &dag.This{Kind: "This"})
```

Reply:
My brain doesn't think that way (you are creating a ref that will be returned and putting it in the symbol table not creating two things that happen to be the same and returning one of them).  I will lift the comment.